src/wp-includes/option.php

```diff
@@ -776,21 +776,13 @@ function update_option( $option, $value, $autoload = null ) {
 	 */
 	$value = apply_filters( 'pre_update_option', $value, $option, $old_value );
 
-	/*
-	 * If the new and old values are the same, no need to update.
-	 *
-	 * Unserialized values will be adequate in most cases. If the unserialized
-	 * data differs, the (maybe) serialized data is checked to avoid
-	 * unnecessary database calls for otherwise identical object instances.
-	 *
-	 * See https://core.trac.wordpress.org/ticket/38903
-	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+	/** This filter is documented in wp-includes/option.php */
+	$default_value = apply_filters( "default_option_{$option}", false, $option, false );
+
+	if ( $old_value !== $default_value && _is_equal_database_value( $old_value, $value ) ) {
 		return false;
-	}
+	} elseif ( $old_value === $default_value ) {
```

Comment:
```suggestion
	} 
	
	if ( $old_value === $default_value ) {
```

The else if doesnt make any sense here, as there is a return on the line above.